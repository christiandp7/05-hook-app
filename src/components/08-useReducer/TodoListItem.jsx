import React from 'react'

const TodoListItem = ({ todo, index: i, handleDelete, handleToggle }) => {
  return (
    <li key={todo.id} className="list-group-item">
      <p 
        className={`mb-0 todo-desc ${todo.done && 'complete'}`}
        onClick={ () => handleToggle(todo.id)}
      >
        {i + 1}. {todo.desc}
      </p>
      <button 
        className="btn btn-danger"
        onClick={ () => handleDelete(todo.id) }
      >
        Borrar
      </button>
    </li>
  )
}

export default TodoListItem
