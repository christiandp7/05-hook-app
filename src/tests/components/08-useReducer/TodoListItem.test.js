import React from 'react'
import { shallow } from 'enzyme'
import TodoListItem from '../../../components/08-useReducer/TodoListItem'
import { demoTodos } from '../../fixtures/demoTodos'

describe('Test <TodoListItem />', () => {
	const handleDelete = jest.fn()
	const handleToggle = jest.fn()

	const wrapper = shallow(
		<TodoListItem
			todo={demoTodos[0]}
			index={0}
			handleDelete={handleDelete}
			handleToggle={handleToggle}
		/>,
	)

	test('should render <TodoListItem />', () => {
		expect(wrapper).toMatchSnapshot()
	})
	test('should call handleDelete function', () => {
		// toHaveBeenCalledWith
		wrapper.find('button').simulate('click')
		expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id)
	})
	test('should call handleToggle function', () => {
		// toHaveBeenCalledWith
		wrapper.find('p').simulate('click')
		expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id)
	})
	test('should show text correctly', () => {
		// p content
		const p = wrapper.find('p')
		expect(p.text().trim()).toBe(`1. ${demoTodos[0].desc}`)
	})
	test('should have complete class when donde is true', () => {
		// p content
		const todo = demoTodos[0]
		todo.done = true
		const wrapper = shallow(
			<TodoListItem
				todo={todo}
				index={0}
				handleDelete={handleDelete}
				handleToggle={handleToggle}
			/>,
		)

		expect(wrapper.find('p').hasClass('complete')).toBe(true)
	})
})
