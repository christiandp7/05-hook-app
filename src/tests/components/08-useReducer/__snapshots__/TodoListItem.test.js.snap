// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TodoListItem /> should render <TodoListItem /> 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="mb-0 todo-desc false"
    onClick={[Function]}
  >
    1
    . 
    Aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
